modules/ROOT/pages/configure-ingress-http-resource.adoc

```diff
@@ -0,0 +1,392 @@
+metadata:  
+  name: example-template-gateway  
+  namespace: rtf  
+spec:  
+  baseEndpoints:  
+    - https://example.com  
+  resources:  
+    - |  
+      apiVersion: gateway.networking.k8s.io/v1  
+      kind: HTTPRoute  
+      metadata:  
+        name: {{ .ResourceName }}  
+        namespace: {{ .Namespace }}  
+      spec:  
+        parentRefs:  
+          - kind: Gateway  
+            name: cluster-gateway  
+        hostnames:  
+          - {{ .Host }}  
+        rules:  
+          - matches:  
+              - path:  
+                  type: PathPrefix  
+                  value: {{ .Path }}  
+            backendRefs:  
+              - name: {{ .Service.Name }}  
+                port: {{ .Service.Port }}  
+----
+
+[NOTE]
+You are responsible for the configuration of the Kubernetes Gateway resources and allowing HTTPRoutes from the route's namespace to be attached.
+
+
+== Using TLS in the HTTP Route-Template Custom Resource
+
+To use TLS in the routing resources in an HTTP route template, you may need to create a TLS secret in the Runtime Fabric namespace. For details, refer to the https://kubernetes.io/docs/concepts/services-networking/ingress/#tls[Kubernetes TLS] documentation.
+
+To configure the Runtime Fabric to manage the secrets lifecycle (create, update, and delete) across all application namespace, use the `rtf.mulesoft.com/synchronized: "true"` label on the TLS secret. For example:
+
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: testsecret-tls
+  namespace: rtf
+  labels:
+    rtf.mulesoft.com/synchronized: "true"
+data:
+  tls.crt: base64 encoded cert
+  tls.key: base64 encoded key
+type: kubernetes.io/tls
+----
+
+Using this label is highly recommended for secrets management. If you don’t add this label, you must manually update the secret in each namespace.
+
+== Modify an Existing HTTP Route-Template CRD
+
+You can modify existing `httproutetemplates` templates at any time. Note that previously created HTTP routing resources based on this template aren’t updated automatically.
+
+To update existing HTTP routing resources after changing the template, perform a redeployment of the Mule app from the Anypoint control plane. For example, if you update the base domain, which requires changing the endpoint for the ingress template, remove the old endpoint and add the new one as part of the application redeployment.
+
+== Create Role and RoleBindings
+
+If the ingress resource is not the standard ingress type, create a Role or RoleBinding to give the Runtime Fabric agent service account the necessary privileges to create an ingress resource.
+
+This example shows the creation of Role and RoleBinding. For Istio, Gateway API, and other similar configurations, create Roles and RoleBindings that are tailored to your specific needs. You can either create these in all application namespaces or use ClusterRole and ClusterRoleBindings, depending on your preferences and any restrictions you have in place.
```

Comment:
Just for consistency, let's add OpenShift route, Gateway API, etc. Instead of Istio, as Istio itself is not an ingress object but it is a service mesh.

Reply:
I dont see our example including roles for GatewayAPI, so i would day lets just keep it for OpenShift route as well

Reply:
@hegderohit89 it is okay to mention gateway api, it will just provide clear idea that it is not just for OpenShift routes.